### PR TITLE
Add tests based on jest

### PR DESCRIPTION
**Pull request by stonko1994**
## Description
This PR will add a basic setup for `jest`.

## Tests
- Content metadata tests (include content metadata overriding)
- Externally session managing tests
- Player events tests (session handling based on player events)

### CODE REVIEW

File: spec/helper/MockHelper.ts [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+
+    return new PlayerMockClass();
+  }
+}
+
+export interface TestingPlayerAPI extends PlayerAPI {
+  eventEmitter: EventEmitter;
+}
+
+interface EventEmitter {
```
Comment (unki2aut) [inline spec/helper/MockHelper.ts:130]: This interface is not really needed, why do we have it?

Comment (stonko1994) [reply spec/helper/MockHelper.ts:130]: This make sure that we don't need an any cast when firing events.

**Review by unki2aut — APPROVED** [review]
Test setup looks good to me :+1:

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  // A map from regular expressions to paths to transformers
+  transform: {
+    '^.+\\.tsx?$': 'ts-jest'
+  },
+
+  // A set of global variables that need to be available in all test environments.
+  globals: {
+    window: { },
+    navigator: { }, // TODO: find a way to mock the BrowserUtils
```
Comment (protyposis) [inline jest.config.js]: Just set `navigator.userAgent` to an appropriate value?

File: spec/helper/PlayerEvent.ts [hunk 3]
```diff
@@ -0,0 +1,807 @@
+export enum PlayerEvent {
```
Comment (protyposis) [inline spec/helper/PlayerEvent.ts:1]: Not a big fan of this. Why can't you use the `PlayerEvent` enum from the player package?

File: spec/tests/ContentMetadata.spec.ts [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+        describe('it does update', () => {
+          it('encoded frame rate', () => {
+            convivaAnalytics.updateContentMetadata({ encodedFrameRate: 144 });
+
+            expect(clientMock.updateContentMetadata).toHaveBeenLastCalledWith(0, expect.objectContaining({
+              encodedFrameRate: 144,
+            }));
+          });
+
+          it('defaultResrouce', () => {
```
Comment (protyposis) [inline spec/tests/ContentMetadata.spec.ts]: Typo

**Review by protyposis — CHANGES_REQUESTED** [review]

File: spec/helper/PlayerEvent.ts (hunk 3 above)
Comment (unki2aut) [reply spec/helper/PlayerEvent.ts:1]: If we import the enum from `bitmovin-player` we can't mock it off and then the whole player code is executed and all the browser specific variables would have to be mocked as well.

We can't import the enum form the types, as `d.ts` files are not compiled by the `tsc`.

Comment (protyposis) [reply spec/helper/PlayerEvent.ts:1]: Sad but true

File: jest.config.js (hunk 2 above)
Comment (stonko1994) [reply jest.config.js]: Found a way to mock it properly.

**Review by protyposis — APPROVED** [review]

**Pull request merged**